src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CreateCaseCallbackService.java

```diff
@@ -163,32 +167,57 @@ private ProcessResult tryCreateNewCase(
-                return ccdNewCaseCreator.createNewCase(
-                    exceptionRecord,
-                    configItem,
-                    ignoreWarnings,
-                    idamToken,
-                    userId,
-                    exceptionRecordData
-                );
-            } else if (ids.size() == 1) {
-                final Map<String, Object> finalizedExceptionRecordData =
-                    exceptionRecordFinalizer.finalizeExceptionRecord(
-                        exceptionRecordData.getData(),
-                        ids.get(0),
-                        CcdCallbackType.CASE_CREATION
-                    );
-                return new ProcessResult(finalizedExceptionRecordData);
-            } else {
+            if (ids.size() > 1) {
                 String msg = String.format(
                     "Multiple cases (%s) found for the given bulk scan case reference: %s",
                     ids.stream().map(String::valueOf).collect(joining(", ")),
                     exceptionRecord.id
                 );
                 log.error(msg);
                 throw new MultipleCasesFoundException(msg);
+            } else {
+                Either<ProcessResult, Long> result = ids
+                    .stream()
+                    .findFirst()
+                    .map(Either::<ProcessResult, Long>right)
+                    .orElseGet(() -> ccdNewCaseCreator.createNewCase(
+                        exceptionRecord,
+                        configItem,
+                        ignoreWarnings,
+                        idamToken,
+                        userId
+                    ));
+
+                Either<ProcessResult, ProcessResult> processedResult = result
+                    .map(id -> {
+                        tryPublishPaymentMessage(configItem.getService(), exceptionRecordData, id);
+
+                        return new ProcessResult(
+                            exceptionRecordFinalizer.finalizeExceptionRecord(
+                                exceptionRecordData.getData(),
+                                id,
+                                CcdCallbackType.CASE_CREATION
+                            )
+                        );
+                    });
+
+                return processedResult.isLeft() ? processedResult.getLeft() : processedResult.get();
             }
         }
     }
+
+    private void tryPublishPaymentMessage(String serviceName, CaseDetails exceptionRecordData, long caseId) {
+        try {
+            paymentsProcessor.updatePayments(exceptionRecordData, caseId);
+        } catch (PaymentsPublishingException exception) {
+            log.error(
+                "Failed to send update to payment processor for {} exception record {}",
+                serviceName,
+                exceptionRecordData.getId(),
+                exception
+            );
+
+            throw new CallbackException("Payment references cannot be processed. Please try again later", exception);
```

Comment:
The caseworker won't see this message - we'll just log the error and return a 500 response. This bit of code has been moved from the other class and it doesn't change the existing behaviour, so not necessarily something for this PR.